sdk/eventhub/Azure.Messaging.EventHubs/design/partition-receiver-proposal.md

```diff
@@ -0,0 +1,451 @@
+﻿# .NET Event Hubs Client: Partition Receiver Proposal
+
+The Event Hubs client library offers two primary clients for consuming events, the `EventHubConsumerClient` and `EventProcessorClient`, each designed for slightly different scenarios but unified in their approach to provide a consistent experience for developers starting with the "Hello World" experience and stepping-up to production use.  These clients embrace a common design philosophy in providing developers an experience optimized around ease of use, familiar patterns, and a consistent API across them.
+
+Because of their respective focus, neither is able to fully address advanced scenarios requiring higher throughput nor where customization is desired.
+
+## Things to know before reading
+
+- The names used in this document are intended for illustration only. Some names are not ideal and may not fully conform to guidelines; these will be refined during discussion and board reviews.
+
+- Some details not related to the high-level concept are not illustrated; the scope of this is limited to the high level shape and paradigms for the feature area.
+
+- Fake methods are used to illustrate "something needs to happen, but the details are unimportant."  As a general rule, if an operation is not directly related to one of the Event Hubs types, it can likely be assumed that it is for illustration only.  These methods will most often use ellipses for the parameter list, in order to help differentiate them.
+
+## Target segment: Developers with specialized needs
+
+These are developers working on products which have special needs that are often specialized and do not fit into the majority case for many Event Hubs client library users. While this segment has a much smaller addressable market, those that fall into this segment often drive a large amount of ACR.
+
+These developers are interested in using the low-level components of the Event Hubs client library, focused around client-service communication that they can control and customize to meet their needs. This class of developers are considered advanced users of Event Hubs with a deep understanding of the service, cloud development, and messaging systems. Many are willing to accept the complexity of working with lower-level components for the ability to have more control for their implementation.
+
+## Why this is needed
+
+Because the `EventHubConsumerClient` is not bound to a specific partition, it cannot directly interact with a partition to read events.  In order to read events from a partition, the client needs to spawn a transport consumer to communicate with the service by creating an AMQP link and manage the state for reading from the partition, such as the identifier of partition and the position wthin the event stream being read.
```

Comment:
```suggestion
Because the `EventHubConsumerClient` is not bound to a specific partition, it cannot directly interact with a partition to read events.  In order to read events from a partition, the client needs to spawn a transport consumer to communicate with the service by creating an AMQP link and manage the state for reading from the partition, such as the identifier of partition and the position within the event stream being read.
```